src/plugins/embeddable/public/lib/panel/panel_header/panel_actions/add_panel/add_panel_flyout.tsx

```diff
@@ -96,41 +105,32 @@ export class AddPanelFlyout extends React.Component<Props> {
     this.showToast(name);
   };
 
-  private getSelectCreateNewOptions() {
-    const list = [
-      {
-        value: 'createNew',
-        inputDisplay: (
-          <EuiText>
-            <FormattedMessage
-              id="embeddableApi.addPanel.createNewDefaultOption"
-              defaultMessage="Create new ..."
-            />
-          </EuiText>
-        ),
-      },
-      ...[...this.props.getAllFactories()]
-        .filter(
-          factory => factory.isEditable() && !factory.isContainerType && factory.canCreateNew()
-        )
-        .map(factory => ({
-          inputDisplay: (
-            <EuiText>
-              <FormattedMessage
-                id="embeddableApi.addPanel.createNew"
-                defaultMessage="Create new {factoryName}"
-                values={{
-                  factoryName: factory.getDisplayName(),
-                }}
-              />
-            </EuiText>
-          ),
-          value: factory.type,
-          'data-test-subj': `createNew-${factory.type}`,
-        })),
-    ];
-
-    return list;
+  private toggleCreateMenu = () => {
+    this.setState(prevState => ({ isCreateMenuOpen: !prevState.isCreateMenuOpen }));
+  };
+
+  private closeCreateMenu = () => {
+    this.setState({ isCreateMenuOpen: false });
+  };
+
+  private getCreateMenuItems() {
+    return [...this.props.getAllFactories()]
+      .filter(factory => factory.isEditable() && !factory.isContainerType && factory.canCreateNew())
+      .map(factory => (
+        <EuiContextMenuItem
+          key={factory.type}
+          data-test-subj={`createNew-${factory.type}`}
+          onClick={() => this.createNewEmbeddable(factory.type)}
+        >
+          <FormattedMessage
+            id="embeddableApi.addPanel.createNew"
+            defaultMessage="Create new {factoryName}"
```

Comment:
Can you just remove the "Create new" portion of this text but make sure that the `factoryName` value's first letter is uppercased?